relayer/types/types.go

```diff
@@ -24,17 +23,17 @@ type Message struct {
 	Data          []byte `json:"data"`
 	MessageHeight uint64 `json:"messageHeight"`
 	EventType     string `json:"eventType"`
+	key           *MessageKey
 }
 
 func (m *Message) MessageKey() *MessageKey {
```

Comment:
Lets not use pointer in MessageKey...as creating pointer and creating newPointer each time will create problem when equating stuct....use pointer only doesn't make code performant....